### PR TITLE
Fix SPARK tech upgrades

### LINKED ISSUE #203: SPARK tech tree issues
It seems as if:

- The MechanizedWarfare tech is not properly available to research in the PG
- SPARKs can be built without the tech being researched
- This leads to having SPARKs whose gear cannot be upgraded
- I also don't have the BIT bombing ability on my SPARKs; may or may not be related

### PR DESCRIPTION
**Pull request by Kazanir**
Fix SPARK tech upgrades via WOTC integrated DLC helper class instead of LW2DLCHelper

Fixes #203.

### CODE REVIEW

**Comment by Kazanir** [comment]
Ok, I believe this will work correctly.

File: LongWarOfTheChosen/Src/LW_Overhaul/Classes/LWDLCHelpers.uc [hunk 1]
```diff
@@ -137,6 +137,21 @@ static function bool IsLostTowersNarrativeEnabled()
 	return false;
 }
 
+// Replaces X2StrategyElement_DLC_Day90Techs.static.IsMechanizedWarfareAvailable()
+static function bool IsMechanizedWarfareAvailable()
+{
+    // We need MechanizedWarfare to be available in Integrated DLC mode. This function replaces the
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_Overhaul/Classes/LWDLCHelpers.uc]: I know it's a trivial thing, but this is going to drive me nuts. A tab here please.

File: LongWarOfTheChosen/Src/LW_Overhaul/Classes/LWTemplateMods.uc [hunk 2]
```diff
@@ -2689,6 +2689,12 @@ function RewireTechTree(X2StrategyElementTemplate Template, int Difficulty)
 		if (TechTemplate.DataName == 'HeavyWeapons') // remove the alternative access to the heavy weapons proving ground project for sparks
 			TechTemplate.AlternateRequirements.Length = 0;
 
+		// Change the special requirements functions for Mechanized Warfare & SPARKs.
+		if (TechTemplate.DataName == 'MechanizedWarfare')
+			TechTemplate.Requirements.SpecialRequirementsFn = class'LWDLCHelpers'.static.IsMechanizedWarfareAvailable;
```
Comment (pledbrook) [inline LongWarOfTheChosen/Src/LW_Overhaul/Classes/LWTemplateMods.uc]: Are you sure about this? Shouldn't the special requirements function be the same as for BuildSpark, i.e. has Lost Towers been completed?

Comment (pledbrook) [reply LongWarOfTheChosen/Src/LW_Overhaul/Classes/LWTemplateMods.uc]: I think this is the right approach, i.e. replacing the SpecialRequirementsFn, I'm just not sure this is the right function. I'm pretty sure the point is for the SpecialRequirementsFn to override the prerequisites, which is ADVENT Robotics for Mechanized Warfare.

**Comment by pledbrook** [comment]
Thanks. Confirmed it works for the three scenarios I mentioned.

**Pull request merged**